### PR TITLE
Add OperatorConfig validation tests

### PR DESCRIPTION
**Pull request by bernot-dev**
Added tests for OperatorConfig validation including mutual exclusivity and edge cases in interval and URL configurations, in response to the user request.

---
*PR created automatically by Jules for task [4736691683968907047](https://jules.google.com/task/4736691683968907047) started by @bernot-dev*

### CODE REVIEW

File: pkg/operator/apis/monitoring/v1/operator_types_test.go [hunk 1]
```diff
@@ -277,6 +277,68 @@ func TestOperatorConfigValidate(t *testing.T) {
...
+											Name: "qux",
+										},
+									},
+								},
+							},
+						}},
+					},
+				},
+			},
+			err: "invalid rules config: invalid alert manager endpoint \x60bar\x60 (index 0): invalid TLS Cert: SecretOrConfigMap fields are mutually exclusive",
```
[the remark this answers is not in the document]
Comment (bernot-dev) [reply pkg/operator/apis/monitoring/v1/operator_types_test.go]: @jules make these changes.

**Comment by bernot-dev** [comment]
@gemini review

**Review by ridwanmsharif — APPROVED** [review]
